src/test/librbd/CMakeLists.txt

```diff
@@ -158,6 +158,9 @@ add_executable(unittest_librbd
   ${unittest_librbd_srcs}
   $<TARGET_OBJECTS:common_texttable_obj>)
 target_compile_definitions(unittest_librbd PRIVATE "TEST_LIBRBD_INTERNALS")
+set_target_properties(unittest_librbd PROPERTIES
```

Comment:
@tchaikov I expect unittest_rbd_mirror may also be a problem.

Reply:
@trociny thanks for your inputs. i agree, we need to tackle this problem using a case-by-case approach. 

but on my building host, the source files included by `rbd_mirror_internal` target take around 800MiB memory on average. in my case:

```
    PID USER      PRI  NI  VIRT   RES   SHR S CPU%-MEM%   TIME+  Command
 657686 kefu       20   0  770M  744M 21612 R 100.  4.7  0:17.44 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/tools/rbd_mirror/CMakeFiles/rbd_mirror_internal.dir/image_replayer/snapshot/Utils.cc.d -MF src/to
 657869 kefu       20   0  801M  775M 20856 R 100.  4.9  0:14.76 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/tools/rbd_mirror/CMakeFiles/rbd_mirror_internal.dir/image_sync/SyncPointCreateRequest.cc.d -MF sr
 658177 kefu       20   0  608M  583M 20364 R 100.  3.7  0:09.91 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/tools/rbd_mirror/CMakeFiles/rbd_mirror_internal.dir/image_sync/SyncPointPruneRequest.cc.d -MF src
 658552 kefu       20   0  347M  318M 16124 R 99.5  2.0  0:03.43 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/tools/rbd_mirror/CMakeFiles/rbd_mirror_internal.dir/pool_watcher/RefreshImagesRequest.cc.d -MF sr
```

and the ones in unittest_rbd_mirror could take up to 1.9 GiB.

```
    PID USER      PRI  NI  VIRT   RES   SHR S CPU%-MEM%   TIME+  Command
 670667 kefu       20   0 1905M 1880M 24184 R 100. 11.8  0:59.42 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/test/rbd_mirror/CMakeFiles/unittest_rbd_mirror.dir/image_deleter/test_mock_SnapshotPurgeRequest.c
 671577 kefu       20   0 1550M 1522M 21080 R 100.  9.5  0:44.40 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/test/rbd_mirror/CMakeFiles/unittest_rbd_mirror.dir/image_deleter/test_mock_TrashMoveRequest.cc.d
 672319 kefu       20   0 1399M 1370M 21136 R 100.  8.6  0:33.00 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/test/rbd_mirror/CMakeFiles/unittest_rbd_mirror.dir/image_deleter/test_mock_TrashWatcher.cc.d -MF
 672136 kefu       20   0 1312M 1280M 21064 R 99.6  8.0  0:35.40 /usr/lib/gcc/x86_64-linux-gnu/11/cc1plus -quiet -I src/include -I ../src -imultiarch x86_64-linux-gnu -MD src/test/rbd_mirror/CMakeFiles/unittest_rbd_mirror.dir/image_deleter/test_mock_TrashRemoveRequest.cc.`
```

so, i am able to use `-j6`on my machine with 16 GB memory without worrying about OOM. for your reference, i am using GCC-11. i understand, your mileage may vary, so if you spot a very high memory usage when compiling `unittest_rbd_mirror`, i will be happy to add add it to the "heavy job" club.

Reply:
No. It is just I know that rbd mirror unit tests are heavily using templates for mock tests, similarly to librbd unit tests, so potentially they may have the same problem. If you think it is still ok I am fine with it too.

Reply:
@tchaikov 
On FreeBSD I seem to be running OOM during linking... 
The VMs have 16Gb of memory, and stiil I manage to get OOM during linking of ceph-dencoder.
So parallelism is already set down to `-j4` to allow building.
A way to specify in cmake that some linkage is to heavy, and should be done at the end, running things serial, would be a great bonus.

Reply:
> A way to specify in cmake that some linkage is to heavy, and should be done at the end, running things serial, would be a great bonus.

@wjwithagen not sure how to archive this. but i will see what i can do to address the linkage issue of `ceph-dencoder`. it is very different, as it includes **lots** of symbols of different structures into a monolithic executable. to address the compile time issue i split a single source file into smaller pieces in eaaec3d3193795822fb220e65a4d18b800d1b95f. but this time, we might need to go further by moving these objects into multiple shared libraries.

Reply:
#40451